gems/aws-sigv4/spec/signer_spec.rb

```diff
@@ -144,6 +144,17 @@ module Sigv4
           expect(signature.headers['host']).to eq('domain.com')
         end
 
+        context 'when a Host header is provided' do
+          
+          let(:headers) { 'host' => 'custom-host-value' }
```

Comment:
It looks like you need to wrap this in an additional {}, ie: `let(:headers) {  { 'host' => 'custom-host-value' } } `